packages/interactive-messages/src/util.ts

```diff
@@ -0,0 +1,62 @@
+import os from 'os';
+import { ErrorCode, errorWithCode } from './errors';
+const pkg = require('../package.json'); // tslint:disable-line
+
+function escape(s: string): string { return s.replace('/', ':').replace(' ', '_'); }
+
+export const errorCodes = {
+  PROMISE_TIMEOUT: ErrorCode.PromiseTimeout,
+};
+
+/**
+ * Creates a timeout on a promise.
+ *
+ * @param ms - The timeout duration.
+ * @param promise - The promise that will timeout.
+ */
+export function promiseTimeout<T>(ms: number, promise: T | Promise<T>): Promise<T> {
+  // Create a promise that rejects in `ms` milliseconds
+  const timeout = new Promise<never>((_resolve, reject) => {
+    const id = setTimeout(
+      () => {
+        clearTimeout(id);
+        reject(errorWithCode(new Error('Promise timed out'), ErrorCode.PromiseTimeout));
+      },
+      ms,
+    );
+  });
+
+  // Race between our timeout and the passed in `promise`
+  return Promise.race([
+    promise as Promise<T>,
+    timeout,
+  ]);
+}
+
+// NOTE: before this can be an external module:
+// 1. are all the JS features supported back to a reasonable version?
+//    default params, template strings, computed property names
+// 2. access to `pkg` will change
+// 3. tests
+// there will potentially be more named exports in this file
+export function packageIdentifier(addons: Record<string, string> = {}): string {
+  const identifierMap = Object.assign(
+    {
+      [pkg.name]: pkg.version,
+      [os.platform()]: os.release(),
+      node: process.version.replace('v', ''),
+    },
+    addons,
+  );
+  return Object.keys(identifierMap).reduce((acc, k) => `${acc} ${escape(k)}/${escape(identifierMap[k])}`, '');
+}
+
+/**
+ * Tests a "thing" for being falsy. See: https://developer.mozilla.org/en-US/docs/Glossary/Falsy
+ *
+ * @param x - The "thing" whose falsy-ness to test.
+ */
+export function isFalsy(x: any): x is 0 | '' | null | undefined {
```

Comment:
out of interest, why? everywhere you use this seems like it'd be happy with `!theThing`. are you trying to work around the type system or something?

Reply:
Mostly working around the linter, actually. It could be better to just disable that rule for this project.

Reply:
which rule was it and whats an example of the line that caused it? there's probably a cleaner solution i suspect

Reply:
For example, the [check inside of `sendResponse`'s return function](https://github.com/clavin/node-slack-sdk/blob/d54766ccc903360c84e6c5bad21b58e6ce959026/packages/events-api/src/http-handler.ts#L63), if it were:

```typescript
if (err) {
  // ...
}
```

Would cause the linter warning:

```
This type is not allowed in the 'if' condition because it could be undefined. Allowed types are boolean or boolean-or-undefined. (strict-boolean-expressions)
```

Reply:
yeah in all honesty its OTT. if my PR merges too, tslint will be removed (as it should be these days) and ESLint doesn't recommend such a rule by default.

i'd recommend you turn the rule off entirely in `tslint.json` for now and drop this helper function. it really is over the top for what is being done here (a check to see if an error is set). if you really wanted to be specific, you'd just do `err !== null && err !== undefined` everywhere but is there really a need?

Reply:
@43081j what you're proposing is a change to the project's style guide. that's welcome! i just don't think this specific PR is the right place to start enforcing that change. i think it would make more sense to first discuss it, and if we all agree on the change then create a new PR which applies it across the codebase.

Reply:
Am I? In the current sources does this helper function exist? I was under the impression it doesn't and had been introduced pretty much to work around the linter or compiler. 

In which case the particular lines triggering the error should really either do a more specific check or should turn the rule off for themselves. 

These functions are typed to accept an error or null usually. It shouldn't be possible to accept some non-null falsy value. Checking `!err` was fine to be honest.

Reply:
> In the current sources does this helper function exist? I was under the impression it doesn't and had been introduced pretty much to work around the linter or compiler.

You're right that this function does not exist previously. But its addition is to help conform the code to satisfy the *linter*, not the typechecker. The linter's job is to enforce the project's style guide.

> In which case the particular lines triggering the error should really either do a more specific check or should turn the rule off for themselves.

As far as I can tell, this function encapsulates what you've called "a more specific check" so that its not repeated several times. The alternative, "turn the rule off" is updating the project's style guide. So I think we agree on this point.

> These functions are typed to accept an error or null usually. It shouldn't be possible to accept some non-null falsy value. Checking `!err` was fine to be honest, or `!== null`

I'm not sure which functions you're referring to. If you mean the conditions where this `isFalsy()` function is being called, then these are not typically functions, they are typically the conditions of an `if` statement.

Whether or not it accepts a non-null falsy value isn't something we can change today without changing the API contract. Even if we agree that the intention is that `null` or `undefined` are the only falsy values we care about, someone who uses this library may not have agreed, and therefore may be triggering a code path in their application that uses some other falsy value. We're trying to apply this change without breaking any existing code, so this function is a useful hygiene utility to accomplish just that.

Reply:
What I mean is the functions which make use of this workaround function are just trying to check if an error is set or some other value is set. 

More than likely you should be checking specifically if the value is null or undefined. That is what the linter really wants, this function is a workaround catch all to pretty much bypass the lint rule IMO.

Reply:
> More than likely you should be checking specifically if the value is null or undefined.

We're intentionally choosing not to depend on any sort of probability. We are trying to produce a change that cannot impact the API of this package in a breaking manner (no matter how badly the user abused the API). This function provides the exact semantics of the "falsy" concept without circumventing the linter or changing the style guide - so it works.

You're right that there are probably places where this gets used that don't need as extensive of a check. But if we're going to encapsulate it in a function for a couple places, we might as well apply that function throughout for consistency (and we get even more confidence that we didn't break the API).

Reply:
If we were open to API breaking changes, I would agree that we should simplify so that this function would become unused and vanish.